variation-commons-core/src/test/java/uk/ac/ebi/eva/commons/core/models/AbstractVariantTest.java

```diff
@@ -1,138 +1,176 @@
+        Variant variant = new Variant("1", 2, 2, "C", "T");
+        variant.renormalize(1, 2, "AC", "AT");
+        assertEquals(1, variant.getStart());
+        assertEquals(2, variant.getEnd());
+    }
+
+    @Test
+    public void testRefAltAssignmentInConstructor() {
+        Variant variant = new Variant("1", 1, 1, "C", "T");
+        assertEquals("C", variant.getReference());
+        assertEquals("T", variant.getAlternate());
+    }
+
+    @Test
+    public void testRefAltAssignmentInRenormalization() {
+        Variant variant = new Variant("1", 1, 1, "", "T");
+        variant.renormalize(1, 2, "A", "AT");
+        assertEquals("A", variant.getReference());
+        assertEquals("AT", variant.getAlternate());
+    }
+
+    @Test
+    public void testNullRefAssignmentInConstructor() {
+        Variant variant = new Variant("1", 1, 1, null, "T");
+        assertEquals("", variant.getReference());
+        assertEquals("T", variant.getAlternate());
+    }
+
+    @Test
+    public void testEmptyRefAssignmentInConstructor() {
+        Variant variant = new Variant("1", 1, 1, "", "T");
+        assertEquals("", variant.getReference());
+        assertEquals("T", variant.getAlternate());
+    }
+
+    @Test
+    public void testNullAltAssignmentInConstructor() {
+        Variant variant = new Variant("1", 1, 1, "T", null);
+        assertEquals("T", variant.getReference());
+        assertEquals("", variant.getAlternate());
+    }
+
+    @Test
+    public void testEmptyAltAssignmentInConstructor() {
+        Variant variant = new Variant("1", 1, 1, "T", "");
+        assertEquals("T", variant.getReference());
+        assertEquals("", variant.getAlternate());
+    }
+
+    @Test
+    public void testGetTypeNoVariant() {
+        assertEquals(VariantType.NO_ALTERNATE, new Variant("1", 1, 1, "T", ".").getType());
+    }
+
+    @Test
+    public void testGetTypeSequenceAlteration() {
+        assertEquals(VariantType.SEQUENCE_ALTERATION, new Variant("1", 1, 1, "", "(100 BP ins)").getType());
+        assertEquals(VariantType.SEQUENCE_ALTERATION, new Variant("1", 1, 1, "T", "(100 BP ins)").getType());
+        assertEquals(VariantType.SEQUENCE_ALTERATION, new Variant("1", 1, 1, "(100 BP del)", "").getType());
+        assertEquals(VariantType.SEQUENCE_ALTERATION, new Variant("1", 1, 1, "(100 BP del)", "T").getType());
+    }
+
+    @Test
+    public void testGetTypeMNV() {
+        assertEquals(VariantType.MNV, new Variant("1", 1, 1, "TAC", "CAT").getType());
+        assertEquals(VariantType.MNV, new Variant("1", 1, 1, "TACGGGG", "CATTTTT").getType());
+    }
+
+    @Test
+    public void testGetTypeSNV() {
+        assertEquals(VariantType.SNV, new Variant("1", 1, 1, "G", "T").getType());
+    }
+
+    @Test
+    public void testGetTypeIndel() {
+        assertEquals(VariantType.INDEL, new Variant("1", 1, 1, "T", "<DEL>").getType());
+        assertEquals(VariantType.INDEL, new Variant("1", 1, 1, "T", "<INS>").getType());
```

Comment:
In reality, we can't really consider angle-bracketed alleles to be INDELs. For instance, in the RS release VCF report we are going to write `<156_DEL_BP>` and consider that a `SEQUENCE_ALTERATION`.

Reply:
I think I agree, have you seen my previous comments in this PR?